### PR TITLE
KAFKA-15102: Add release notes about the replication.policy.internal.topic.separator.enabled property for MirrorMaker 2

### PR DESCRIPTION
**Pull request by C0urante**
[Jira](https://issues.apache.org/jira/browse/KAFKA-15102)

This is a docs-only follow-up for https://github.com/apache/kafka/pull/14082 that calls out the necessity for this property in our release notes so that users upgrading MM2 from older versions can do so safely.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by C0urante** [comment]
@gharris1727 @mimaison as the other two committers who voted for KIP-949, would you mind taking a look? Thanks!

File: docs/upgrade.html
Comment (C0urante) [inline docs/upgrade.html:49]: We can also link to [KIP-949](https://cwiki.apache.org/confluence/display/KAFKA/KIP-949%3A+Add+flag+to+enable+the+usage+of+topic+separator+in+MM2+DefaultReplicationPolicy) here, but I'm hoping that the information provided here and in the property's docstring should be sufficient to save people the time and effort of reading through an entire KIP instead.

Comment (gharris1727) [inline docs/upgrade.html:49]: This second link feels redundant.

Comment (gharris1727) [reply docs/upgrade.html:49]: It does look like the other items in this section include KIP links, but in general there aren't too many KIP links in the upgrade section. I think linking the KIP is unnecessary.

**Review by gharris1727 — COMMENTED** [review]
I think that we should copy this line item to the 3.1.x-3.5.x documentation when the patch release for those versions go out, but we might be in danger of forgetting. Can you create blocker tickets with fix versions for the future patch releases that will remind us to document this?

File: docs/upgrade.html
Comment (C0urante) [reply docs/upgrade.html:49]: Is there a disadvantage to it, though? Making it easier to reach the docs (even for a small fraction of users) is a clear benefit; I'm not sure there's a significant cost to providing two links instead of one.

**Comment by C0urante** [comment]
> I think that we should copy this line item to the 3.1.x-3.5.x documentation when the patch release for those versions go out, but we might be in danger of forgetting. Can you create blocker tickets with fix versions for the future patch releases that will remind us to document this?

Based on our [release policy](https://cwiki.apache.org/confluence/display/KAFKA/Time+Based+Release+Plan#TimeBasedReleasePlan-WhatIsOurEOLPolicy?), it looks like 3.1 and 3.2 are EOL, so it's unlikely we'll need to backport any further than 3.3.

I'm also planning on opening a separate PR to backport this to 3.5 (which can then be cherry-picked to 3.4 and 3.3) once this PR is approved. I'd just cherry-pick the commit for this PR, but we don't have autogenerated docs for the MM2 properties on those branches, so the wording will have to be different. Once those backport commits land, they should be picked up and included in the next releases on those branches automatically.

This will still omit the notes on, e.g., upgrading to 3.4.2, from the main docs page at https://kafka.apache.org/documentation/, but it seems like this is a general issue with our docs. For example, our current trunk's [upgrade notes](https://github.com/apache/kafka/blob/f970ddff10114ae7f9ea1b9966cd25ac8bb5f581/docs/upgrade.html) don't include a section for 3.5.1, even though one is included in the [upgrade notes](https://github.com/apache/kafka/blob/abd1c8e46fc413f8cb8d2d07a80ee74eaf5d9708/docs/upgrade.html) for our current 3.5 branch). Do you think it's worth making a special case for this PR, or is it acceptable to leave upgrade information on older versions (e.g., 3.4.2) off of the docs for newer versions (e.g., 3.5.2)?

File: docs/upgrade.html
Comment (gharris1727) [reply docs/upgrade.html:49]: It may just be my personal preference then. I'm used to the "wikipedia style" (i don't know what else to call it) where a link is provided the first time something is mentioned and then not afterwards. When I see two links adjacent to one another, I expect that they will go somewhere different.

This is not a blocking issue.

**Comment by gharris1727** [comment]
> I'd just cherry-pick the commit for this PR, but we don't have autogenerated docs for the MM2 properties on those branches, so the wording will have to be different.

Yes the backport PRs will need to be different, if only because the 3.6.0 section won't exist. I was thinking that in addition to this upgrade note, we would also need to backport the configuration documentation to keep the links valid, and because the documentation would be incomplete without the new configuration present. That would need a delayed-merge PR against kafka-site instead of merging to kafka today.

> This will still omit the notes on, e.g., upgrading to 3.4.2, from the main docs page at https://kafka.apache.org/documentation/, but it seems like this is a general issue with our docs. For example, our current trunk's [upgrade notes](https://github.com/apache/kafka/blob/f970ddff10114ae7f9ea1b9966cd25ac8bb5f581/docs/upgrade.html) don't include a section for 3.5.1, even though one is included in the [upgrade notes](https://github.com/apache/kafka/blob/abd1c8e46fc413f8cb8d2d07a80ee74eaf5d9708/docs/upgrade.html) for our current 3.5 branch). Do you think it's worth making a special case for this PR, or is it acceptable to leave upgrade information on older versions (e.g., 3.4.2) off of the docs for newer versions (e.g., 3.5.2)?

I agree that seems like a general issue, and I do not think we need to address it here. It looks like the "upgrade to 3.4.2" doesn't already exist and we would need to create it, but we should discuss with others on whether it is appropriate to cherry-pick that section to 3.5/trunk.

**Comment by C0urante** [comment]
> I was thinking that in addition to this upgrade note, we would also need to backport the configuration documentation to keep the links valid, and because the documentation would be incomplete without the new configuration present. That would need a delayed-merge PR against kafka-site instead of merging to kafka today.

Ah, I see. I wasn't planning on backporting the complete config docs (IMO it's too high-risk to backport the config autogeneration logic), I was planning on tweaking the [older docs on MM2's properties](https://github.com/apache/kafka/blob/abd1c8e46fc413f8cb8d2d07a80ee74eaf5d9708/docs/ops.html#L768-L788) and linking to those in the release notes. I've filed https://github.com/apache/kafka/pull/14235 to demonstrate.

@gharris1727 since I've created the backport PR, do you think it would be safe to merge this PR (assuming wording and formatting are satisfactory) and continue discussing a backport strategy on the backport PR?

File: docs/upgrade.html
Comment (C0urante) [reply docs/upgrade.html:49]: > When I see two links adjacent to one another, I expect that they will go somewhere different.

That's a fair point. I'm a little hesitant to only include a link in the first mention of the property, since people may still not think to click the property to see docs for it. I've modified the first mention of it to use a `<code>` block instead, and kept the second link where it was. How does that look to you?

**Review by gharris1727 — APPROVED** [review]
> do you think it would be safe to merge this PR (assuming wording and formatting are satisfactory) and continue discussing a backport strategy on the backport PR?

Yes that is a good idea.
LGTM!

**Pull request merged**
